From b39fc6efda5ff1f76342dd75b3a4ba474a8be809 Mon Sep 17 00:00:00 2001
From: pexcn <pexcn97@gmail.com>
Date: Mon, 4 Aug 2025 15:46:22 +0800
Subject: [PATCH] remove unwanted languages

Signed-off-by: pexcn <pexcn97@gmail.com>
---
 config/common.mk | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/config/common.mk b/config/common.mk
index e8109c6..316e36e 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -315,13 +315,6 @@ PRODUCT_PACKAGES += \
     NetworkStackOverlay \
     PermissionControllerOverlay
 
-# Translations
-CUSTOM_LOCALES += \
-    ast_ES \
-    gd_GB \
-    cy_GB \
-    fur_IT
-
 PRODUCT_ENFORCE_RRO_EXCLUDED_OVERLAYS += vendor/crowdin/overlay
 PRODUCT_PACKAGE_OVERLAYS += vendor/crowdin/overlay
 
-- 
2.45.2

