From 8e618bdb7b38eec70556eb6c2d3982cbfd5ed3f0 Mon Sep 17 00:00:00 2001
From: minaripenguin <minaripenguin@users.noreply.github.com>
Date: Thu, 12 Sep 2024 22:09:13 +0800
Subject: [PATCH] MobileSignalController: Prevent possible memory leak

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 .../statusbar/connectivity/MobileSignalController.java     | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/connectivity/MobileSignalController.java b/packages/SystemUI/src/com/android/systemui/statusbar/connectivity/MobileSignalController.java
index b495657560c6..fc2b22e987e6 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/connectivity/MobileSignalController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/connectivity/MobileSignalController.java
@@ -87,6 +87,7 @@ public class MobileSignalController extends SignalController<MobileState, Mobile
     boolean mInflateSignalStrengths = false;
     @VisibleForTesting
     final MobileStatusTracker mMobileStatusTracker;
+    private final TunerService mTunerService;
 
     // Save the previous STATUS_HISTORY_SIZE states for logging.
     private final String[] mMobileStatusHistory = new String[STATUS_HISTORY_SIZE];
@@ -170,8 +171,7 @@ public class MobileSignalController extends SignalController<MobileState, Mobile
         };
         mMobileStatusTracker = mobileStatusTrackerFactory.createTracker(mMobileCallback);
 
-        Dependency.get(TunerService.class).addTunable(this, DATA_DISABLED_ICON);
-        Dependency.get(TunerService.class).addTunable(this, SHOW_FOURG_ICON);
+        mTunerService = Dependency.get(TunerService.class);
     }
 
     @Override
@@ -234,6 +234,8 @@ public class MobileSignalController extends SignalController<MobileState, Mobile
         mContext.getContentResolver().registerContentObserver(Global.getUriFor(
                 Global.MOBILE_DATA + mSubscriptionInfo.getSubscriptionId()),
                 true, mObserver);
+        mTunerService.addTunable(this, DATA_DISABLED_ICON);
+        mTunerService.addTunable(this, SHOW_FOURG_ICON);
     }
 
     /**
@@ -242,6 +244,7 @@ public class MobileSignalController extends SignalController<MobileState, Mobile
     public void unregisterListener() {
         mMobileStatusTracker.setListening(false);
         mContext.getContentResolver().unregisterContentObserver(mObserver);
+        mTunerService.removeTunable(this);
     }
 
     private void updateInflateSignalStrength() {
-- 
2.49.0

