From 8d1b8484202f203fccd1cc9ed5dfd6c6f49afb11 Mon Sep 17 00:00:00 2001
From: pexcn <pexcn97@gmail.com>
Date: Fri, 26 Sep 2025 11:45:20 +0800
Subject: [PATCH] recovery: allow SPL downgrade on user builds

Signed-off-by: pexcn <pexcn97@gmail.com>
---
 install/install.cpp |  6 ------
 recovery.cpp        | 16 ++++++----------
 2 files changed, 6 insertions(+), 16 deletions(-)

diff --git a/install/install.cpp b/install/install.cpp
index d898a93..06babc8 100644
--- a/install/install.cpp
+++ b/install/install.cpp
@@ -426,14 +426,8 @@ static InstallResult TryUpdateBinary(Package* package, bool* wipe_cache,
   bool device_supports_virtual_ab = android::base::GetBoolProperty("ro.virtual_ab.enabled", false);
 
   bool spl_downgrade_approved = false;
-  const auto allow_spl_downgrade =
-      android::base::GetBoolProperty("persist.vendor.recovery_allow_spl_downgrade", false);
   const auto current_spl = android::base::GetProperty("ro.build.version.security_patch", "");
   if (ViolatesSPLDowngrade(zip, current_spl)) {
-    if (!allow_spl_downgrade || !ui->IsTextVisible()) {
-      LOG(ERROR) << "Denying OTA because it's SPL downgrade";
-      return INSTALL_ERROR;
-    }
     if (!ask_to_continue_spl_downgrade(device)) {
       LOG(ERROR) << "User denied SPL downgrade";
       return INSTALL_ERROR;
diff --git a/recovery.cpp b/recovery.cpp
index c255431..56684af 100644
--- a/recovery.cpp
+++ b/recovery.cpp
@@ -188,16 +188,12 @@ bool ask_to_continue_downgrade(Device* device) {
 }
 
 bool ask_to_continue_spl_downgrade(Device* device) {
-  if (get_build_type() == "user") {
-    return false;
-  } else {
-    device->GetUI()->SetProgressType(RecoveryUI::EMPTY);
-    return yes_no(device,
-      "WARNING: Security patch level downgrade detected. "
-      "This may require formatting data. "
-      "Device may brick if hardware rollback protection is enabled. ",
-      "Install anyway?");
-  }
+  device->GetUI()->SetProgressType(RecoveryUI::EMPTY);
+  return yes_no(device,
+    "WARNING: Security patch level downgrade detected. "
+    "This may require formatting data. "
+    "Device may brick if hardware rollback protection is enabled. ",
+    "Install anyway?");
 }
 
 static bool ask_to_wipe_data(Device* device) {
-- 
2.45.2

