From 6cf8a9c38d2f4ecacb417a39e8d88c44cd93df1b Mon Sep 17 00:00:00 2001
From: pexcn <pexcn97@gmail.com>
Date: Tue, 30 Sep 2025 23:33:35 +0800
Subject: [PATCH 2/2] overlay: remove keybox overlay

---
 config/evolution.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config/evolution.mk b/config/evolution.mk
index 7b702a6..babbe5c 100644
--- a/config/evolution.mk
+++ b/config/evolution.mk
@@ -16,7 +16,6 @@ PRODUCT_BUILD_PROP_OVERRIDES += \
 
 # Evolution X packages
 PRODUCT_PACKAGES += \
-    CertifiedKeyboxOverlay \
     EvoEgg \
     GameSpace \
     OmniJaws \
-- 
2.50.1

